doc/kernel/usermode/syscalls.rst

```diff
@@ -304,17 +324,15 @@ information is not necessary since all arguments and the return value are
         ...
     }
 
-Note that system calls may have more than six arguments. In this case,
-the sixth and subsequent arguments to the system call are placed into a struct,
-and a pointer to that struct is passed to the handler as its sixth argument.
-See ``include/syscall.h`` to see how this is done; the struct passed in must be
-validated like any other memory buffer.
-
 After validating all the arguments, the handler function needs to then call
 the implementation function. If the implementation function returns a value,
 this needs to be returned by the handler, otherwise the handler should return
 0.
 
+Do not forget that all the arguments to the handler are passed in as unsigned
```

Comment:
This looks important.  How about:

    .. note::  All arguments to the handler are passed in as unsigned
       32-bit values.  If checks are needed...